app/tests/integration/nimbus/utils/filter_expression.js

```diff
@@ -0,0 +1,25 @@
+Components.utils.import("resource://gre/modules/components-utils/FilterExpressions.jsm");
+Components.utils.import("resource://gre/modules/components-utils/ClientEnvironment.jsm");
+
+// Targeting
+
+async function remoteSettings(arguments) {
+
+    const TargetingContext = ChromeUtils.import(
+        "resource://messaging-system/targeting/Targeting.jsm"
+    )
+
+    _experiment = JSON.parse(arguments[1])
+    const targetingContext = new TargetingContext.TargetingContext(_experiment);
+    let result = false;
+    try {
+        result = await targetingContext.evalWithDefault(arguments[0]);
```

Comment:
What will this return? A Boolean `true` or `false` or throw an error which gets caught below on L17?

Reply:
Overall I think it's better that we're using this custom code and not calling into `evaluateJexl` directly.
This way we can catch the error and return it to python for a better error message.

> What will this return? A Boolean true or false or throw an error which gets caught below on L17?

Depends on the argument `a == 1` will return `true/false`, something like `true && foo` where `foo` is an undefined variable will return `undefined` or that variables value. Any syntax errors: typos or unknown functions will throw. It should match [our docs](https://experimenter.info/desktop-targeting-debug#targeting).